### PR TITLE
refactor: minor improve and remove unused traits impls for f32/f64

### PR DESCRIPTION
**Pull request by yjhmelody**
And do code fmt for traits impl orders.

### CODE REVIEW

File: crates/wasmi/src/lib.rs [hunk 1]
```diff
@@ -86,7 +86,7 @@
 #[macro_use]
 extern crate alloc;
 #[cfg(feature = "std")]
-extern crate std as alloc;
+extern crate alloc;
```
Comment (Robbepop) [inline crates/wasmi/src/lib.rs:89]: This is just wrong. We want to use the `std` lib if the user specifies to do so.

Comment (yjhmelody) [reply crates/wasmi/src/lib.rs:89]: But no code actually use `std` alias. I consider it a code style but forget to hold on it.

**Comment by paritytech-cicd-pr** [comment]
## BENCHMARKS ## 

 <table> <thead> <tr> <th></th><th colspan=3>NATIVE</th><th colspan=3>WASMTIME</th><th></th> </tr> <tr> <th>BENCHMARK</th><th>MASTER</th><th>PR</th><th>DIFF</th><th>MASTER</th><th>PR</th><th>DIFF</th><th>WASMTIME OVERHEAD</th> </tr> </thead> <tbody> <tr><td><tt>execute/</tt><br><tt>bare_call_0</td> <td nowrap> 1.27ms</td> <td nowrap> 1.30ms</td> <td nowrap> :red_circle: 1.88%</td> <td nowrap> 1.12ms</td> <td nowrap> 1.09ms</td> <td nowrap> :green_circle: -3.00%</td> <td nowrap> :green_circle: -16%</td></tr><tr><td><tt>execute/</tt><br><tt>bare_call_0/typed</td> <td nowrap> 710.72µs</td> <td nowrap> 749.99µs</td> <td nowrap> :red_circle: 6.28%</td> <td nowrap> 541.65µs</td> <td nowrap> 540.81µs</td> <td nowrap> :white_circle: -0.24%</td> <td nowrap> :green_circle: -28%</td></tr><tr><td><tt>execute/</tt><br><tt>bare_call_1</td> <td nowrap> 1.39ms</td> <td nowrap> 1.42ms</td> <td nowrap> :red_circle: 2.58%</td> <td nowrap> 1.35ms</td> <td nowrap> 1.35ms</td> <td nowrap> :green_circle: -0.10%</td> <td nowrap> :green_circle: -5%</td></tr><tr><td><tt>execute/</tt><br><tt>bare_call_16</td> <td nowrap> 2.97ms</td> <td nowrap> 3.00ms</td> <td nowrap> :white_circle: 1.80%</td> <td nowrap> 5.00ms</td> <td nowrap> 5.00ms</td> <td nowrap> :green_circle: 0.09%</td> <td nowrap> :yellow_circle: 67%</td></tr><tr><td><tt>execute/</tt><br><tt>bare_call_16/typed</td> <td nowrap> 1.29ms</td> <td nowrap> 1.30ms</td> <td nowrap> :white_circle: 0.97%</td> <td nowrap> 2.31ms</td> <td nowrap> 2.28ms</td> <td nowrap> :green_circle: -1.76%</td> <td nowrap> :yellow_circle: 75%</td></tr><tr><td><tt>execute/</tt><br><tt>bare_call_1/typed</td> <td nowrap> 800.88µs</td> <td nowrap> 852.73µs</td> <td nowrap> :red_circle: 6.55%</td> <td nowrap> 869.94µs</td> <td nowrap> 864.92µs</td> <td nowrap> :white_circle: -0.81%</td> <td nowrap> :green_circle: 1%</td></tr><tr><td><tt>execute/</tt><br><tt>bare_call_4</td> <td nowrap> 1.78ms</td> <td nowrap> 1.80ms</td> <td nowrap> :white_circle: 2.12%</td> <td nowrap> 2.16ms</td> <td nowrap> 2.16ms</td> <td nowrap> :green_circle: 0.12%</td> <td nowrap> :green_circle: 20%</td></tr><tr><td><tt>execute/</tt><br><tt>bare_call_4/typed</td> <td nowrap> 893.27µs</td> <td nowrap> 887.96µs</td> <td nowrap> :white_circle: -0.65%</td> <td nowrap> 998.64µs</td> <td nowrap> 982.08µs</td> <td nowrap> :green_circle: -1.69%</td> <td nowrap> :green_circle: 11%</td></tr><tr><td><tt>execute/</tt><br><tt>br_table</td> <td nowrap> 938.17µs</td> <td nowrap> 952.10µs</td> <td nowrap> :red_circle: 1.41%</td> <td nowrap> 1.10ms</td> <td nowrap> 1.09ms</td> <td nowrap> :white_circle: -0.21%</td> <td nowrap> :green_circle: 15%</td></tr><tr><td><tt>execute/</tt><br><tt>count_until</td> <td nowrap> 1.20ms</td> <td nowrap> 903.17µs</td> <td nowrap> :green_circle: -24.64%</td> <td nowrap> 2.21ms</td> <td nowrap> 2.21ms</td> <td nowrap> :white_circle: -0.03%</td> <td nowrap> :red_circle: 144%</td></tr><tr><td><tt>execute/</tt><br><tt>factorial_iterative</td> <td nowrap> 365.05µs</td> <td nowrap> 364.89µs</td> <td nowrap> :white_circle: 0.00%</td> <td nowrap> 903.19µs</td> <td nowrap> 902.98µs</td> <td nowrap> :white_circle: -0.09%</td> <td nowrap> :red_circle: 147%</td></tr><tr><td><tt>execute/</tt><br><tt>factorial_recursive</td> <td nowrap> 711.40µs</td> <td nowrap> 710.85µs</td> <td nowrap> :white_circle: -0.08%</td> <td nowrap> 1.44ms</td> <td nowrap> 1.44ms</td> <td nowrap> :white_circle: -0.02%</td> <td nowrap> :red_circle: 103%</td></tr><tr><td><tt>execute/</tt><br><tt>fib_iterative</td> <td nowrap> 1.79ms</td> <td nowrap> 1.79ms</td> <td nowrap> :white_circle: -0.02%</td> <td nowrap> 4.91ms</td> <td nowrap> 4.91ms</td> <td nowrap> :white_circle: -0.23%</td> <td nowrap> :red_circle: 174%</td></tr><tr><td><tt>execute/</tt><br><tt>fib_recursive</td> <td nowrap> 6.75ms</td> <td nowrap> 6.76ms</td> <td nowrap> :white_circle: 0.15%</td> <td nowrap> 13.52ms</td> <td nowrap> 13.58ms</td> <td nowrap> :white_circle: 0.38%</td> <td nowrap> :red_circle: 101%</td></tr><tr><td><tt>execute/</tt><br><tt>global_bump</td> <td nowrap> 1.38ms</td> <td nowrap> 1.38ms</td> <td nowrap> :white_circle: 0.13%</td> <td nowrap> 3.36ms</td> <td nowrap> 3.35ms</td> <td nowrap> :white_circle: -0.28%</td> <td nowrap> :red_circle: 143%</td></tr><tr><td><tt>execute/</tt><br><tt>global_const</td> <td nowrap> 999.52µs</td> <td nowrap> 1.00ms</td> <td nowrap> :white_circle: 0.10%</td> <td nowrap> 2.51ms</td> <td nowrap> 2.52ms</td> <td nowrap> :white_circle: 0.16%</td> <td nowrap> :red_circle: 151%</td></tr><tr><td><tt>execute/</tt><br><tt>host_calls</td> <td nowrap> 32.38µs</td> <td nowrap> 32.38µs</td> <td nowrap> :white_circle: 0.02%</td> <td nowrap> 44.69µs</td> <td nowrap> 44.74µs</td> <td nowrap> :white_circle: 0.10%</td> <td nowrap> :green_circle: 38%</td></tr><tr><td><tt>execute/</tt><br><tt>memory_fill</td> <td nowrap> 1.55ms</td> <td nowrap> 1.55ms</td> <td nowrap> :white_circle: 0.05%</td> <td nowrap> 4.08ms</td> <td nowrap> 4.07ms</td> <td nowrap> :white_circle: -0.26%</td> <td nowrap> :red_circle: 162%</td></tr><tr><td><tt>execute/</tt><br><tt>memory_sum</td> <td nowrap> 1.55ms</td> <td nowrap> 1.55ms</td> <td nowrap> :white_circle: -0.03%</td> <td nowrap> 4.06ms</td> <td nowrap> 4.05ms</td> <td nowrap> :white_circle: -0.08%</td> <td nowrap> :red_circle: 161%</td></tr><tr><td><tt>execute/</tt><br><tt>memory_vec_add</td> <td nowrap> 3.06ms</td> <td nowrap> 4.89ms</td> <td nowrap> :red_circle: 59.70%</td> <td nowrap> 8.13ms</td> <td nowrap> 8.03ms</td> <td nowrap> :white_circle: -1.01%</td> <td nowrap> :yellow_circle: 64%</td></tr><tr><td><tt>execute/</tt><br><tt>recursive_is_even</td> <td nowrap> 1.30ms</td> <td nowrap> 1.29ms</td> <td nowrap> :white_circle: -0.82%</td> <td nowrap> 2.57ms</td> <td nowrap> 2.49ms</td> <td nowrap> :green_circle: -3.23%</td> <td nowrap> :yellow_circle: 93%</td></tr><tr><td><tt>execute/</tt><br><tt>recursive_ok</td> <td nowrap> 166.97µs</td> <td nowrap> 167.24µs</td> <td nowrap> :white_circle: 0.08%</td> <td nowrap> 343.74µs</td> <td nowrap> 345.02µs</td> <td nowrap> :white_circle: 0.17%</td> <td nowrap> :red_circle: 106%</td></tr><tr><td><tt>execute/</tt><br><tt>recursive_scan</td> <td nowrap> 209.01µs</td> <td nowrap> 213.20µs</td> <td nowrap> :red_circle: 2.03%</td> <td nowrap> 439.92µs</td> <td nowrap> 438.20µs</td> <td nowrap> :white_circle: -0.48%</td> <td nowrap> :red_circle: 106%</td></tr><tr><td><tt>execute/</tt><br><tt>recursive_trap</td> <td nowrap> 16.42µs</td> <td nowrap> 16.25µs</td> <td nowrap> :white_circle: -1.29%</td> <td nowrap> 34.30µs</td> <td nowrap> 33.93µs</td> <td nowrap> :white_circle: -1.15%</td> <td nowrap> :red_circle: 109%</td></tr><tr><td><tt>execute/</tt><br><tt>regex_redux</td> <td nowrap> 678.60µs</td> <td nowrap> 664.76µs</td> <td nowrap> :green_circle: -2.03%</td> <td nowrap> 1.45ms</td> <td nowrap> 1.45ms</td> <td nowrap> :white_circle: -0.13%</td> <td nowrap> :red_circle: 118%</td></tr><tr><td><tt>execute/</tt><br><tt>rev_complement</td> <td nowrap> 583.99µs</td> <td nowrap> 583.93µs</td> <td nowrap> :white_circle: 0.01%</td> <td nowrap> 1.43ms</td> <td nowrap> 1.43ms</td> <td nowrap> :white_circle: -0.04%</td> <td nowrap> :red_circle: 146%</td></tr><tr><td><tt>execute/</tt><br><tt>tiny_keccak</td> <td nowrap> 462.06µs</td> <td nowrap> 475.46µs</td> <td nowrap> :red_circle: 2.96%</td> <td nowrap> 1.29ms</td> <td nowrap> 1.32ms</td> <td nowrap> :red_circle: 2.64%</td> <td nowrap> :red_circle: 178%</td></tr><tr><td><tt>execute/</tt><br><tt>trunc_f2i</td> <td nowrap> 1.01ms</td> <td nowrap> 1.02ms</td> <td nowrap> :red_circle: 1.46%</td> <td nowrap> 2.45ms</td> <td nowrap> 2.44ms</td> <td nowrap> :white_circle: -0.28%</td> <td nowrap> :red_circle: 138%</td></tr><tr><td><tt>instantiate/</tt><br><tt>wasm_kernel</td> <td nowrap> 72.09µs</td> <td nowrap> 71.16µs</td> <td nowrap> :white_circle: 0.25%</td> <td nowrap> 94.88µs</td> <td nowrap> 94.64µs</td> <td nowrap> :white_circle: -0.28%</td> <td nowrap> :green_circle: 33%</td></tr><tr><td><tt>translate/</tt><br><tt>wasm_kernel</td> <td nowrap> 4.18ms</td> <td nowrap> 4.17ms</td> <td nowrap> :white_circle: -0.10%</td> <td nowrap> 8.03ms</td> <td nowrap> 7.79ms</td> <td nowrap> :green_circle: -3.46%</td> <td nowrap> :yellow_circle: 87%</td></tr> </tbody> </table> 

 [Link to pipeline](https://gitlab.parity.io/parity/mirrors/wasmi/-/jobs/1998695)

**Review by Robbepop — APPROVED** [review]
LGTM Thanks for the follow-up.

**Pull request merged**